src/docs/Dockerfile

```diff
@@ -0,0 +1,40 @@
+## Base ########################################################################
+# Use a larger node image to do the build for native deps (e.g., gcc, python)
+FROM node:lts as base
+
+# Reduce npm log spam and colour during install within Docker
+ENV NPM_CONFIG_LOGLEVEL=warn
+ENV NPM_CONFIG_COLOR=false
+
+# We'll run the app as the `node` user, so put it in their home directory
+WORKDIR /home/node/app
+# Install pnpm
+RUN npm install -g pnpm
+# Copy the source code over
+COPY --chown=node:node . /home/node/app/
+
+## Development #################################################################
+# Define a development target that installs devDeps and runs in dev mode
+FROM base as development
+WORKDIR /home/node/app
+# Install (not ci) with dependencies, and for Linux vs. Linux Musl (which we use for -alpine)
+RUN pnpm install
+# Switch to the node user vs. root
+USER node
+# Expose port 4631
+EXPOSE 4631
+# Start the app in debug mode so we can attach the debugger
+CMD ["pnpm", "start"]
+
+## Production ##################################################################
+# Also define a production target which doesn't use devDeps
+FROM base as production
+WORKDIR /home/node/app
+COPY --chown=node:node --from=development /home/node/app/node_modules /home/node/app/node_modules
+
+## Deploy ######################################################################
+# Use a stable nginx image
+FROM nginx:stable-alpine as deploy
```

Comment:
I'm curious about this: are we going to deploy another instance of `nginx` or are we going to use the one serving our main pages to serve the docs?

Reply:
I'm not entirely sure how we'd do this with a single instance from a Docker point of view.  We have totally separate packages for `src/web` and `src/docs`.  Without merging them, how should we share the server?

I agree, having a single static host makes a lot of sense.  But we could move to that in a follow-up.

Reply:
Discussing this with @manekenpix, in a follow-up we could do:

```
src/web/Dockerfile
src/web/app
src/web/docs
```

Now we can build both static sites (next.js, Docusaurus) and host them on the same nginx instance.

Any Docker folks want to try this one?

Reply:
Filed https://github.com/Seneca-CDOT/telescope/issues/2993 for later.

Reply:
Moving Docusaurus into /web/ makes so much sense! We should also rename the `/docs` directory as well. I realized some of the file paths currently look like `src/docs/docs/*` and that will likely be prone to errors.

I will make a comment on the follow-up issue above.